src/azure-cli/azure/cli/command_modules/monitor/operations/general_operations.py

```diff
@@ -0,0 +1,62 @@
+from .._client_factory import cf_monitor
+from ..util import _gen_guid
+from azure.cli.core.commands.transform import _parse_id
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.tools import parse_resource_id
+
+
+logger = get_logger(__name__)
+
+
+def clone_existed_settings(cmd, source_resource=None, target_resource=None):
+    metrics_alert_rules = _clone_monitor_metrics_alerts(cmd, source_resource, target_resource)
+    return metrics_alert_rules
+
+
+def _clone_monitor_metrics_alerts(cmd, source_resource, target_resource):
+    if not _is_resource_type_same_and_sub_same(source_resource, target_resource):
+        raise CLIError('The target resource should be the same type with the source resource '
+                       'and they are in the same subscription')
+    monitor_client = cf_monitor(cmd.cli_ctx, subscription_id=parse_resource_id(source_resource)['subscription'])
+    # we can only clone the alert rules belonging to the source subscription.
+    alert_rules = list(monitor_client.metric_alerts.list_by_subscription())
+    updated_alert_rules = []
+    ErrorResponseException = cmd.get_models('ErrorResponseException', operation_group='metric_alerts')
+    for alert_rule in alert_rules:
+        if source_resource in alert_rule.scopes:
+            if target_resource not in alert_rule.scopes:
+                try:
+                    alert_rule.scopes.append(target_resource)
+                    resource_group_name, name = _parse_id(alert_rule.id).values()
+                    alert_rule = monitor_client.metric_alerts.create_or_update(resource_group_name=resource_group_name,
+                                                                               rule_name=name, parameters=alert_rule)
+                except ErrorResponseException as ex:  # Create new alert rule
+                    if ex.response.status_code == 400:
+                        alert_rule.scopes = [target_resource]
+                        resource_group_name, name = _parse_id(target_resource).values()
+                        name = "{}-{}".format(name, _gen_guid())
+                        alert_rule = monitor_client.metric_alerts.create_or_update(resource_group_name=resource_group_name,  # pylint: disable=line-too-long
+                                                                                   rule_name=name,
+                                                                                   parameters=alert_rule)
+                    else:
+                        raise ex
+                updated_alert_rules.append(alert_rule)
+            else:
+                logger.warning('The target resource already has alert rule %s. '
+                               'Skip cloning this one.', alert_rule.name)
+    if not updated_alert_rules:
+        return None
+    return updated_alert_rules if len(updated_alert_rules) > 1 else updated_alert_rules[0]
```

Comment:
>updated_alert_rules [](start = 11, length = 19)

Why do you return different type : list, object, None. Is this CLI's recommendated output?

Reply:
I don't find any existed guideline for such scenario. I think it make senses that we only return list.(list, list with only one object, empty list). It's just a little bit strange for me to have an empty list. I will change the return type to list to keep consistence.